src/mlpack/methods/kmeans/kmeans_impl.hpp

```diff
@@ -228,11 +229,17 @@ Cluster(const MatType& data,
       {
         Log::Info << "Cluster " << i << " is empty.\n";
         if (iteration % 2 == 0)
-          emptyClusterAction.EmptyCluster(data, i, centroids, centroidsOther,
+        {
+          numberOfRemovedClusters = emptyClusterAction.EmptyCluster(data, i, centroids, centroidsOther,
               counts, metric, iteration);
+          clusters = clusters - numberOfRemovedClusters;
```

Comment:
Hmm, I think we can actually simplify this code a bit.  If we change the loop

```
for (size_t i = 0; i < clusters; i++)
```

to

```
for (size_t i = 0; i < counts.n_elem; ++i)
```

then there should be no need for `numberOfRemovedClusters` and `EmptyCluster()` can still return `void`.  Let me know what you think.

Reply:
Indeed its a good suggestion. I've simplified the code!